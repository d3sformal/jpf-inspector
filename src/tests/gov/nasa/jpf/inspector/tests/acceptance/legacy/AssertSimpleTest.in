assert *AssertSimpleTest.java:9 #stackFrame[0].i < 3

run
# Assert should trigger twice:
  # Iteration:
  wait
  thread_pc 0
  print #stackFrame[0].i
  run
  # Iteration:
  wait
  thread_pc 0
  print #stackFrame[0].i
  run
wait
# This should fail because we have alrady terminated:
print #thread