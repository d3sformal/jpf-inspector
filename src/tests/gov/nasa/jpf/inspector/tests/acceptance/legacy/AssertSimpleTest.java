package gov.nasa.jpf.inspector.tests.acceptance.legacy;

import gov.nasa.jpf.vm.Verify;

public class AssertSimpleTest {
  public static void main(String[] args) {
    int i = Verify.getInt(0, 4);
    System.out.println(i);
    return; // <- must be at line 9
  }
}
