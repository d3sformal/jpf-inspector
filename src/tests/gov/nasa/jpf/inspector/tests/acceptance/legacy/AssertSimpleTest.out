New assertion successfully created with ID 1.
INFO: JPF created and connected, SuT is started
INFO: Assertion violated: [assert *AssertSimpleTest.java:9 (#thread.#stackFrame[0].i < 3)]
0 : gov/nasa/jpf/inspector/tests/acceptance/legacy/AssertSimpleTest.java:9:    return; // <- must be at line 9
	gov.nasa.jpf.inspector.tests.acceptance.legacy.AssertSimpleTest:main:7:return  gov.nasa.jpf.inspector.tests.acceptance.legacy.AssertSimpleTest.main([Ljava/lang/String;)V
	1 : ???-RawView-no Name or Type provided (int) = 3
INFO: Assertion violated: [assert *AssertSimpleTest.java:9 (#thread.#stackFrame[0].i < 3)]
0 : gov/nasa/jpf/inspector/tests/acceptance/legacy/AssertSimpleTest.java:9:    return; // <- must be at line 9
	gov.nasa.jpf.inspector.tests.acceptance.legacy.AssertSimpleTest:main:7:return  gov.nasa.jpf.inspector.tests.acceptance.legacy.AssertSimpleTest.main([Ljava/lang/String;)V
	1 : ???-RawView-no Name or Type provided (int) = 4
INFO: JPF Terminating
ERR: This command may only be used when JPF has started and is paused. If JPF is not already started, then start it using 'run' and if it is already running, then pause it using 'break'. Then you will be able to execute this command. Alternatively, you may disable safe mode in the configuration file by setting 'jpf-inspector.safe-mode = false'.